src/psyclone/nemo.py

```
@@ -0,0 +1,822 @@
+        #   Loop variable, [start value expression, end value expression, step
+        #   expression]
+        # Loop variable will be an instance of Fortran2003.Name
+        loop_var = str(ctrl[0].items[1][0])
+        self._variable_name = str(loop_var)
+
+        # Identify the type of loop
+        if self._variable_name in NEMO_LOOP_TYPE_MAPPING:
+            self.loop_type = NEMO_LOOP_TYPE_MAPPING[self._variable_name]
+        else:
+            self.loop_type = "unknown"
+
+        # Get the loop limits. These are given in a list which is the second
+        # element of a tuple which is itself the second element of the items
+        # tuple:
+        # (None, (Name('jk'), [Int_Literal_Constant('1', None), Name('jpk'),
+        #                      Int_Literal_Constant('1', None)]), None)
+        limits_list = ctrl[0].items[1][1]
+        self._start = str(limits_list[0])
+        self._stop = str(limits_list[1])
+        if len(limits_list) == 3:
+            self._step = str(limits_list[2])
+        else:
+            # Default loop increment is 1
+            self._step = "1"
+
+        # Is this loop body a kernel?
+        if NemoKern.match(self._ast):
+            self.addchild(NemoKern(self._ast, parent=self))
+            return
+        # It's not - walk on down the AST...
+        self.process_nodes(self, self._ast.content, self._ast)
+
+    def __str__(self):
+        result = ("NemoLoop[" + self._loop_type + "]: " + self._variable_name +
+                  "=" + ",".join([self._start, self._stop, self._step]) + "\n")
+        for entity in self._children:
+            result += str(entity) + "\n"
+        result += "EndLoop"
+        return result
+
+    @property
+    def kernel(self):
+        ''' Returns the kernel object if one is associated with this loop,
+        None otherwise. '''
+        kernels = self.walk(self.children, NemoKern)
+        if kernels:
+            # TODO cope with case where loop contains >1 kernel (e.g.
```

We should raise an exception here, not just blindly continue!